### PR TITLE
Move windows CI builds from appveyor to Travis CI.

### PR DESCRIPTION
**Pull request by BenjaminW3**
Implements #685

Notes:
* I had to set the `CC` and `CXX` environment variables within the `before_install.sh` (executed in the git bash) again because those are set to g++/gcc even though they are already overwritten in the .travis.yml file. This may be a problem with the current Travis CI windows environment and may be fixed later.
* We do not yet explicitly set the 64 bit compiler build so the compilation may be done by the 32 bit compiler (both can produce 32 and 64 bit results)

TODO:
* [x] merge #714 and rebase
* [x] delete the appveyor project/account (after this has been merged)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -190,23 +209,24 @@ matrix:
...
 cache:
     directories:
         - $ALPAKA_CI_DOCKER_CACHE_DIR
 
 script:
     - set -eovx pipefail
-    - sudo apt-get -y --quiet --allow-unauthenticated --no-install-recommends install smem
-    - sudo apt-get -y --quiet --allow-unauthenticated --no-install-recommends install moreutils
-    - ./script/travis/script.sh | ts
+    - if [ "$TRAVIS_OS_NAME" = "linux" ] ;then sudo apt-get -y --quiet --allow-unauthenticated --no-install-recommends install smem ;fi
```
Comment (ax3l) [inline .travis.yml:226]: you can make this in one line: `... install smem moreutils`

Comment (BenjaminW3) [reply .travis.yml:226]: I know. They have been added at different times and I still want to move the installation of `smem` to the place where it is really needed.

**Comment by BenjaminW3** [comment]
Ready to be merged!

**Review by psychocoderHPC — APPROVED** [review]

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
Will merge, the failing AppVeyor makes sense since it is removed now.

**Pull request merged**